src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -0,0 +1,492 @@
+                private void AnalyzeDelegateCreationOrAnonymousFunction(OperationAnalysisContext operationAnalysisContext)
+                    => _hasDelegateCreationOrAnonymousFunction = true;
+
+                private void AnalyzeConversion(OperationAnalysisContext operationAnalysisContext)
+                {
+                    var conversion = (IConversionOperation)operationAnalysisContext.Operation;
+                    if (!_hasConversionFromDelegateTypeToNonDelegateType &&
+                        conversion.Operand.Type.IsDelegateType() &&
+                        !conversion.Type.IsDelegateType())
+                    {
+                        _hasConversionFromDelegateTypeToNonDelegateType = true;
+                    }
+                }
+
+                private void AnalyzeFieldOrPropertyReference(OperationAnalysisContext operationAnalysisContextContext)
+                {
+                    var fieldOrPropertyReference = operationAnalysisContextContext.Operation;
+                    if (!_delegateAssignedToFieldOrProperty &&
+                        fieldOrPropertyReference.Type.IsDelegateType() &&
+                        fieldOrPropertyReference.Parent is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target == fieldOrPropertyReference)
+                    {
+                        _delegateAssignedToFieldOrProperty = true;
+                    }
+                }
+
+                /// <summary>
+                /// Method invoked in <see cref="AnalyzeOperationBlockEnd(OperationBlockAnalysisContext)"/>
+                /// for each operation block to determine if we should analyze the operation block or bail out.
+                /// </summary>
+                private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol)
+                {
+                    switch (operationBlock.Kind)
+                    {
+                        case OperationKind.None:
+                        case OperationKind.ParameterInitializer:
+                            // Skip blocks from attributes (which have OperationKind.None) and parameter initializers.
+                            // We don't have any unused values in such operation blocks.
+                            return false;
+                    }
+
+                    // We currently do not support points-to analysis, so we cannot accurately 
+                    // track delegate invocations for all cases.
+                    // We attempt to do our best effort delegate invocation analysis as follows:
+
+                    //  1. If we have no delegate creations or lambdas, our current analysis works fine,
+                    //     return true.
+                    if (!_hasDelegateCreationOrAnonymousFunction)
+                    {
+                        return true;
+                    }
+
+                    //  2. Bail out if we have a delegate escape via an assigment to a field/property reference.
```

Comment:
is it explained *why* we're bailing out here?  maybe i missed it.

Reply:
Added a better comment.

---
In reply to: [234803349](https://github.com/dotnet/roslyn/pull/30777#discussion_r234803349) [](ancestors = 234803349)

Reply:
thanks!